gateway/engine/policies/src/test/java/io/apiman/gateway/engine/policies/BasicAuthLDAPSTest.java

```diff
@@ -292,7 +290,8 @@ public PolicyFailure createFailure(PolicyFailureType type, int failureCode, Stri
         });
 
         // The LDAP stuff we're testing!
-        Mockito.when(context.getComponent(ILdapComponent.class)).thenReturn(new DefaultLdapComponent());
+        //DefaultLdapComponentReloaded inherited class is used to load in sslcontext the new truststore defined in setUp()
+        Mockito.when(context.getComponent(ILdapComponent.class)).thenReturn(new DefaultLdapComponentReloaded());
```

Comment:
Just to see if I'm understanding this code correctly:

Was the issue that you were  only loading the SSL context once, so you were unable to switch certificate sets?

Reply:
Exactly!

There are many Policies tests that use DefaultLdapComponent(). The first test, using DefaultLdapComponent, load the static protected ssl context.

So, if BasicAuthLDAPSTest is not run the first, the custom truststore define in his @Before is not used. DefaultLdapComponent can't trust Apache DS LDAP Directory. The test fail.

I don't see this problem before because for some strange reason, maven test don't run tests in the same order on my workstation and on our CI.

With DefaultLdapComponentReloaded class, we could test DefaultLdapComponent features with a reset ssl context.

BasicAuthLDAPSTest will run successfully whatever test order.

I hope this fix will suit you.